### ISSUE TITLE
Dropdown Menu content not working with Apple Pencil / iPad

### ISSUE THREAD

**Issue by dreamorosi** [radix-ui/primitives#1027]
## Bug report

While using Apple Pencil on iOS 15 for iPad I'm not able to interact with the content of the dropdown menu which doesn't register at all those interactions.

### Current Behavior
In the video below I try to perform the same interactions first via regular tap (first 5 seconds of video) and then again with the pencil, which only opens the menu but doesn't register other events.

https://user-images.githubusercontent.com/7353869/145498003-0a93df38-1517-42f6-9851-5ba52eda115a.MOV

### Expected behavior

Consistent behaviour between taps and pencil taps. Other components like slider, switch, etc seem to work fine.

### Reproducible example

Example from the docs:
https://www.radix-ui.com/docs/primitives/components/dropdown-menu

### Suggested solution

Register all taps made with pencil in the same way as other tap events

### Additional context

N/A

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |         |         |
| React            | n/a     |         |
| Browser          |         Safari |         |
| Assistive tech   |         |         |
| Node             | n/a     |         |
| npm/yarn         |         |         |
| Operating System |         iOS |     15    |

**Comment by benoitgrelard** [comment]
Thanks for that, this is rather bizarre.
I don't have an iPad myself but I will try and get access to one.

**Comment by dreamorosi** [comment]
Thanks, happy to test again and/or provide logs if needed

**Comment by alex-sherwin** [comment]
I tried to figure this out for a few hours tonight (using a iPad + Apple Pencil)

I haven't found the root cause as of yet (the radix-ui event handling seems to have a surprising amount of complexity)

But, what I can say is that, the best I can tell, that this only occurs when a `<DropdownMenu>` is rendered in modal mode (the default), when it's non-modal mode, it always works as expected.

What I can observe is that when a `<DropdownMenu>` is rendered in modal mode, when using a Apple Pencil to touch a given `<DropdownMenuItem>`, the `onclick` event handler for the rendered `<div role="menuitem">` doesn't fire (never, even if I hack in a onclick handler function using dev tools, it doesn't fire)

So my assumption is that something relating to the modal behavior is cancelling/preventing the onclick behavior for the `DropdownMenuItem` from executing

Other events on the `<div role="menuitem">` do fire, such as `onpointerdown` and `onpointerup`, which are `PointerEvent` instances where `PointerEvent.pointerType` is one of *touch* (finger), *pen* (apple pencil) or *mouse*.  These events always fire (in modal mode as well).

If you have any suggestions on relevant places to look/debug, I'd be happy to keep looking into this.

My selfish motivation here is to find a solution for https://github.com/tldraw/tldraw/issues/380

**Comment by benoitgrelard** [comment]
Thanks for the extra feedback @alex-sherwin 🙏

**Comment by steveruizok** [comment]
Strange, the `DialogClose` button works correctly with iPad pencil in dialogs.

**Comment by teofum** [comment]
I ran into this issue and have been looking into it a bit. To add to what's been said, I logged all pointer events to see what actually changed between a finger touch and pencil touch, and found the pencil triggers several `pointermove` events in between `pointerdown` and `pointerup`, likely because of the increased precision detecting any tiny movement. As @alex-sherwin mentioned the `click` event never fires for the pencil, either.

Tapping with a finger and moving it slightly to trigger a move event also causes the click not to fire, and I can get the click to fire with the pencil by doing a very quick tap that doesn't trigger any move events, which leads me to believe those move events are what's preventing the click from firing. Moving the pointer while down does still trigger a click with a mouse, though.

So, to sum up what we know:
- The issue happens in any dropdown, dialog, or popover with `modal=true`, which would point to the common component in these which is the focus trapping.
- A `click` event never fires, causing the `onSelect` event not to fire as well.
- The click doesn't fire only when the pointer moves in between up and down events, for both `pen` and `touch` pointer types. Does work with a mouse, so I would look at any code where behavior differs for touch vs mouse events.

I'll keep looking into this and see if I can find a root cause.
